SteakBot.Core/Modules/QuoteModule.cs

```diff
@@ -11,62 +11,251 @@ namespace SteakBot.Core.Modules
+
+            var message = await channel.GetMessageAsync(messageId);
+            var embed = CreateEmbed(message);
+            await SendQuote(Context.User, embed);
+        }
+
+        [Command("quote")]
+        [Summary("Quotes a group of messages specified by Message ID of the first and last messages." +
+                 "Only takes messages that belong to the author of the first one.")]
+        public async Task Quote(string channelMention, ulong firstMessageId, ulong lastMessageId)
+        {
+            if (!TryGetChannel(channelMention, out var channel))
+            {
+                await ReplyAsync("Unknown channel specified.");
+                return;
+            }
+
             await Context.Channel.DeleteMessageAsync(Context.Message, RequestOptions.Default);
 
-            var users = GetUsers().ToList();
+            var messages = await GetMessageList(channel, firstMessageId, lastMessageId);
+            var embed = CreateEmbed(messages);
+            await SendQuote(Context.User, embed);
+        }
+
+        [Command("quote")]
+        [Summary("Quotes a single message specified by Message URI.")]
+        public async Task Quote(Uri messageUri)
+        {
+            var parts = messageUri.AbsolutePath.Split('/').Reverse().ToArray();
+
+            if (!ulong.TryParse(parts[0], out var messageId) ||
+                !ulong.TryParse(parts[1], out var channelId) ||
+                !ulong.TryParse(parts[2], out var guildId))
+            {
+                return;
+            }
+
+            if (!TryGetGuild(guildId, out var guild))
+            {
+                return;
+            }
+
+            if (!TryGetChannel(guild, channelId, out var channel))
+            {
+                return;
+            }
+
+            await Context.Channel.DeleteMessageAsync(Context.Message, RequestOptions.Default);
+
+            var message = await channel.GetMessageAsync(messageId);
+            var embed = CreateEmbed(message);
+            await SendQuote(Context.User, embed);
+        }
+
+        [Command("quote")]
+        [Summary("Quotes a group of messages specified by Message URI of the first and last messages." +
+                 "Only takes messages that belong to the author of the first one.")]
+        public async Task Quote(Uri firstMessageUri, Uri lastMessageUri)
+        {
+            var parts = firstMessageUri.AbsolutePath.Split('/').Reverse().ToArray();
+
+            if (!ulong.TryParse(parts[0], out var firstMessageId) ||
+                !ulong.TryParse(parts[1], out var channelId) ||
+                !ulong.TryParse(parts[2], out var guildId))
+            {
+                return;
+            }
+
+            if (!TryGetGuild(guildId, out var guild))
+            {
+                return;
+            }
+
+            if (!TryGetChannel(guild, channelId, out var channel))
+            {
+                return;
+            }
```

Comment:
Maybe combine these two ifs into one since they thematically fall under the same step in the flow.